### DISCUSSION TITLE
ЛР 6 для ИВТ. Как проверять работоспособность подсистемы прерывания?

### DISCUSSION THREAD

**Discussion by BROsandr** [MPSU/exAPS#44] [Вопросы и ответы] [answered]
Может посоветовать, как проверять прерывания( инструкции CSR, mret и т.д. )? Мы нашли заметку про RARS из 4ой лабы.

> [RARS](https://github.com/TheThirdOne/rars) - еще один инструмент, тоже отрисована микроархитектура, но написан на Java, поэтому не должно быть проблем с запуском на любой операционной системе. Надо только [Java-машину](https://www.java.com/ru/download/ie_manual.jsp?locale=ru) поставить, конечно же. Еще ОБРАТИ ВНИМАНИЕ он умеет работать с инструкциями для CSR, которые используются в прерываниях ([лабораторная 6](https://github.com/MPSU/APS/blob/%D0%98%D0%92%D0%A2/Labs/6.%20Interrupt%20subsystem/README.md)). Скачиваем [тут](https://github.com/TheThirdOne/rars/releases/tag/continuous)

Но можете поподробней написать про тестирование процессора, потому что про это в лабе ни слова? До этого мы пользовались venus, и он ни csr, ни mret не поддерживает.

**Comment by gr33nka** [comment, marked as answer]
К сожалению, ручками. То есть нужно написать программку на языке ассемблера, например, в RARS, запустить ее на симулированном процессоре и потом смотреть потактово, что все работает правильно и в соответствии с ожиданиями. Например, написали программку: вечный цикл + обработчик прерывания на 10 входе прерывания (подобно примеру из лабника). Потом запускаете и смотрите, что цикл шлепает и сам он в прерывание не заходит. Потом моделируете сигнал, что прерывание произошло на 5 входе - ничего не происходит, потом на 10-ом - происходит прерывание, значит правильно работает. Типа такого.

Да, это плохая верификация, она не гарантирует правильный результат, плюс неудобная, но с одной стороны альтернатив пока нет, а с другой это некоммерческое изделие, поэтому можно)

@HepoH3 может быть добавит слова о автоматической проверке, в прошлом году ребята поднимали сервис для этого.

Резюмирую: пишите программу, смотрите поведение на временных диаграммах. Если поведение прогнозируемое, то все работает правильно. + ждем комментарий от Андрея Павловича

**Reply by HepoH3** [reply]
Не, Михаил Геннадьевич все крайне подробно описал, я в общем-то на лабораторной тоже самое проговорил, забыл тут продублировать. Автоматизированная проверка в этом семестре походу не заведется, так что пока проверяем все самостоятельно.
